core/src/main/java/io/questdb/std/Os.java

```diff
@@ -125,6 +126,19 @@ public static int setCurrentThreadAffinity(int cpu) {
         return setCurrentThreadAffinity0(cpu);
     }
 
+    public static void sleep(long millis) {
+        final long t = System.currentTimeMillis();
+        long deadline = millis;
+        while (deadline > 0) {
+            try {
+                Thread.sleep(deadline);
+                break;
+            } catch (InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)